vllm/v1/kv_cache_interface.py

```diff
@@ -76,6 +77,36 @@ def bytes_for_tokens(self, num_tokens: int) -> int:
         return cdiv(num_tokens, self.block_size) * self.page_size_bytes
 
 
+@dataclass
+class SlidingWindowSpec(KVCacheSpec):
+    num_kv_heads: int
+    head_size: int
+    dtype: torch.dtype
+    sliding_window: int
+    use_mla: bool
+
+    @property
+    def type_id(self) -> str:
+        return f"sliding_window_{self.sliding_window}_{self.block_size}_{self.page_size_bytes}"  # noqa
+
+    @property
+    def page_size_bytes(self) -> int:
+        # Sliding window does not affect the page size, so reuse the one for
+        # full attention.
+        # Skip the type check due to mypy bug
+        # https://github.com/python/mypy/issues/8085
+        return FullAttentionSpec.page_size_bytes.fget(  # type: ignore
+            cast(FullAttentionSpec, self))
+
+    def bytes_for_tokens(self, num_tokens: int) -> int:
```

Comment:
QQ: How do we calculate the memory need to support `max_model_len` with SWA?
As written in the PR description, I guess it's more complicated than `bytes_for_tokens(max_model_len)`. Wondering if this complexity is already considered.